quic/s2n-quic-transport/src/endpoint/connection_close.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::endpoint;
+use alloc::collections::VecDeque;
+use core::ops::Range;
+use s2n_codec::{DecoderBufferMut, EncoderValue};
+use s2n_quic_core::{
+    connection,
+    crypto::{InitialHeaderKey, InitialKey},
+    event,
+    frame::ConnectionClose,
+    inet::ExplicitCongestionNotification,
+    io::tx,
+    packet::{initial::CleartextInitial, number::PacketNumberSpace},
+    path::{self, MINIMUM_MAX_DATAGRAM_SIZE},
+    time, transport,
+    varint::VarInt,
+};
+
+#[derive(Debug)]
+pub struct Dispatch<Path: path::Handle> {
+    transmissions: VecDeque<Transmission<Path>>,
+}
+
+impl<Path: path::Handle> Default for Dispatch<Path> {
+    fn default() -> Self {
+        Self::new(endpoint::DEFAULT_MAX_PEERS)
+    }
+}
+
+impl<Path: path::Handle> Dispatch<Path> {
+    pub fn new(max_peers: usize) -> Self {
+        Self {
+            transmissions: VecDeque::with_capacity(max_peers),
```

Comment:
It's a bit unfortunate that we're adding a ~1MB allocation to all endpoints, even those that don't make use of this feature.

(1MB comes from DEFAULT_MAX_PEERS = 1024 and `Transmission` being around 1300 bytes, which means this is ~1.3MB of allocation).

Maybe we could avoid storing encoded packets or figure out a smaller size than 1200 bytes for the payload?